libs/nest/core/src/lib/decorators/nationalId.decorator.ts

```diff
@@ -8,19 +8,19 @@ import { isPerson } from 'kennitala'
 
 export const NationalId = createParamDecorator(
   (_: unknown, context: ExecutionContext): string => {
-    const headers = context.switchToHttp().getRequest().headers
+    const request = context.switchToHttp().getRequest()
+    const nationalId = request.headers['x-param-nationalid']
 
-    const nationalId = headers['x-param-nationalid']
     if (!nationalId) {
       throw new BadRequestException('No national id provided')
     }
 
-    if (isPerson(nationalId) && nationalId.length === 10) {
-      return nationalId
+    if (!isPerson(nationalId) && nationalId.length !== 10) {
```

Comment:
Hoare logic bug
```suggestion
    if (!isPerson(nationalId) || nationalId.length !== 10) {
```

Reply:
that's not De Morgans law, there are no unions there 🤷🏼 

but I am not going to argue about this.

Reply:
Whatever the name, since you're flipping the conditional (throw on invalid instead of return on valid), the `&&` should be `||

Reply:
Well, you namedropped, causing confusion :)